dallinger/command_line/docker_ssh.py

```diff
@@ -363,6 +396,11 @@ def deploy(
     dashboard_user = config.get("dashboard_user", "admin")
     dashboard_password = config.get("dashboard_password", secrets.token_urlsafe(8))
 
+    recruiter = by_name(config["recruiter"])
+    recruiter.validate_config(
```

Comment:
This `validate_config` logic needs to be called also on non-Docker-SSH deployments

Reply:
This issue was about Docker SSh initially, but it seems grew into something bigger.

The `cli_invocation_options` are now passed to `run_pre_launch_checks`. I also updated the Docker and non-Docker deploy commands for the `--open-recruitment` flag.

There is more refactoring that could be made, but we should maybe not include it in this PR.

Reply:
Well, since this PR now includes the implementation of `validate_config`, I think it's important that this implementation be complete rather than partial. At the moment it looks like a bug that this `validate_config` is only called on docker-ssh deployments.

Reply:
Actually, looking at the code I see you've fixed this already.